### ISSUE TITLE
Add a CI job that tests `forc init` and that generated code builds and tests successfully

### ISSUE THREAD

**Issue by mitchmindtree** [FuelLabs/sway#1346]
I'm imagining a new, very basic CI job (under `.github/workflows/ci.yml`) that does the following:

1. Install forc.
2. Runs `forc init test-proj`
3. Runs `forc build --path test-proj`
4. Runs `forc test --path test-proj`

We should possibly also run steps 2 to 4, but using each of the examples as a template, so e.g. loop over each example and replace step 2 with `forc init --template <example> test-proj`.

Something along these lines should ensure the most common first impressions of Sway/Fuel stay shiny and working!

**Comment by adlerjohn** [comment]
Related: #1136

**Comment by ra0x3** [comment]
@mitchmindtree What do you think about me picking this up? Seems not "super necessary" at the moment, but also seems like it could be a good way to get familiar with some things including forc and CI. I do agree this is a great way to make sure everything reliably works e2e (so to speak)

**Comment by mitchmindtree** [comment]
Yeah this would be a great candidate and would help a bunch! Feel free to assign yourself when ready to pick it up.
